python/lsst/meas/base/base.py

```diff
@@ -105,14 +125,19 @@ def register(self, name, PluginClass):
         The name will be used as a prefix for all fields produced by the Plugin, and it
         should generally contain the name of the Plugin or Algorithm class itself
         as well as enough of the namespace to make it clear where to find the code.
+
+        Set canApCorr True if this algorithm measure a flux that can be aperture corrected;
+        the name will be added to the set retrieved by getApCorrNameSet.
         """
         lsst.pex.config.Registry.register(self, name, self.Configurable(name, PluginClass))
+        if canApCorr:
+            addApCorrName(name)
```

Comment:
While this mechanism for registering aperture-corrected names is what we want for most algorithms, it should be a convenience option that delegates to a more general mechanism that allows a single algorithm to register multiple flux fields to be corrected.  The CModel algorithm in `meas_modelfit` has three flux fields, for instance: `modelfit_CModel_flux`, `modelfit_CModel_exp_flux`, and `modelfit_CModel_dev_flux`.

The documentation should also state explicitly that any name added to the registry is the prefix for a trio of `{name}_flux`, `{name}_fluxSigma`, and `{name}_flag` fields.